packages/web/src/lib/core/auth/internal/StitchAuthImpl.ts

```diff
@@ -56,6 +85,129 @@ export default class StitchAuthImpl extends CoreStitchAuth<StitchUser>
     );
   }
 
+  public loginWithRedirect(credential: StitchRedirectCredential) {
+    const { redirectUrl, state } = this.prepareRedirect(
+      credential,
+      RedirectTypes.Login
+    );
+
+    window.location.replace(
+      this.browserAuthRoutes.getAuthProviderRedirectRoute(
+        credential,
+        redirectUrl,
+        state,
+        this.deviceInfo
+      )
+    );
+  }
+
+  public linkUserWithRedirect(
+    user: StitchUser,
+    credential: StitchRedirectCredential
+  ) {
+    this.storage.set(
+      RedirectKeys.LinkUser,
+      JSON.stringify(new StitchUserCodec(this).encode(user))
+    );
+
+    const { redirectUrl, state } = this.prepareRedirect(
+      credential,
+      RedirectTypes.Login
+    );
+
+    const link = this.browserAuthRoutes.getAuthProviderLinkRedirectRoute(
+      credential,
+      redirectUrl,
+      state,
+      this.deviceInfo
+    );
+
+    fetch(
+      new Request(link, {
+        credentials: "include",
+        headers: {
+          Authorization: "Bearer " + this.authInfo.accessToken
+        }
+      })
+    ).then(response => {
```

Comment:
[minor] Hmm, since this could fail, should we return a Promise from linkUserWithRedirect? otherwise this is an unhandled promise rejection that the user can't catch.